ecosystem/typescript/sdk/src/transaction_builder/builder.ts

```diff
@@ -103,3 +129,150 @@ export class TransactionBuilderMultiEd25519 extends TransactionBuilder<SigningFn
     return bcsToBytes(this.signInternal(rawTxn));
   }
 }
+
+/**
+ * Config for creating raw transactions.
+ */
+interface ABIBuilderConfig {
+  sender: HexString | AccountAddress;
+  sequenceNumber: Uint64 | string;
+  gasUnitPrice?: Uint64 | string;
+  maxGasAmount?: Uint64 | string;
+  expSecFromNow?: number | string;
+  chainId: Uint8 | string;
+}
+
+/**
+ * Builds raw transactions based on ABI
+ */
+export class TransactionBuilderABI {
+  private readonly abiMap: Map<string, ScriptABI>;
+
+  private readonly builderConfig: ABIBuilderConfig;
+
+  /**
+   * Constructs a TransactionBuilderABI instance
+   * @param abis List of binary ABIs.
+   * @param builderConfig Configs for creating a raw transaction.
+   */
+  constructor(abis: Bytes[], builderConfig: ABIBuilderConfig) {
+    this.abiMap = new Map<string, ScriptABI>();
+
+    abis.forEach((abi) => {
+      const deserializer = new Deserializer(abi);
+      const scriptABI = ScriptABI.deserialize(deserializer);
+      let k: string;
+      if (scriptABI instanceof ScriptFunctionABI) {
+        const funcABI = scriptABI as ScriptFunctionABI;
+        const { address: addr, name: moduleName } = funcABI.module_name;
+        k = `${HexString.fromUint8Array(addr.address).toShortString()}::${moduleName.value}::${funcABI.name}`;
+      } else {
+        const funcABI = scriptABI as TransactionScriptABI;
+        k = funcABI.name;
+      }
+
+      if (this.abiMap.has(k)) {
+        throw new Error("Found conflicting ABI interfaces");
+      }
+
+      this.abiMap.set(k, scriptABI);
+    });
+
+    this.builderConfig = {
+      gasUnitPrice: 1n,
+      maxGasAmount: 1000n,
+      expSecFromNow: 10,
+      ...builderConfig,
+    };
+  }
+
+  private static toBCSArgs(abiArgs: any[], args: any[]): Bytes[] {
```

Comment:
is this abiArgs only used as type_tag array. If so, should we just called it type_tags: any[]

Reply:
`abiArgs`'s type is ArgumentABI, not TypeTag.

Reply:
You are right. Let me rename